stan/math/prim/mat/fun/Eigen.hpp

```diff
@@ -1,7 +1,7 @@
 #ifndef STAN_MATH_PRIM_MAT_FUN_EIGEN_HPP
 #define STAN_MATH_PRIM_MAT_FUN_EIGEN_HPP
 
-#define EIGEN_MATRIXBASE_PLUGIN "stan/math/prim/mat/eigen_plugins.hpp"
+#define EIGEN_MATRIXBASE_PLUGIN "stan/math/prim/mat/eigen_plugins.h"
```

Comment:
Something I just thought of. Here should we add a check to see if `EIGEN_MATRIXBASE_PLUGIN` is already defined and if so throw a warning to the user? I think that would help our users who add their own extensions know what to do when they want their own plugins

Reply:
That's a good idea, what do you think of:
```
#if defined EIGEN_MATRIXBASE_PLUGIN
#error Please use the eigen_plugins.h header to add your own extensions
#else
#define EIGEN_MATRIXBASE_PLUGIN "stan/math/prim/mat/eigen_plugins.h"
#endif
```

Reply:
I'd rather not have someone add a file to the library. Here is what I'm thinking

Add this to our plugin at the bottom.

```cpp
#define EIGEN_STAN_MATRIXBASE_PLUGIN 
```

Then we have the check look like

```cpp
#ifdef EIGEN_MATRIXBASE_PLUGIN
  #ifndef EIGEN_STAN_MATRIXBASE_PLUGIN
  #error "Stan uses Eigen's EIGEN_MATRIXBASE_PLUGIN macro. To use your own plugin add 
    the eigen_plugin.h file to your plugin. 
  #endif
#else 
#define EIGEN_MATRIXBASE_PLUGIN "stan/math/prim/mat/eigen_plugins.h"
#endif
```

Then users can go about writing their own plugin and only need to include the `eigen_plugins.h` file in their plugin. 

We can also then move the stuff in `autodiff_variables.h` back to `eigen_plugins.h`

Reply:
Oh I see what you mean, that would be a bit tidier. Good call